src/otx/engine/utils/hpo/__init__.py

```diff
@@ -0,0 +1,9 @@
+# Copyright (C) 2024 Intel Corporation
```

Comment:
Is the difference between `otx/hpo` and `engine/utils/hpo` just to distinguish the functions used by engine?
If so, wouldn't it be better to put them all in one file called `utils/hpo.py or hpo_utils.py`?

Reply:
I don't think so. At first, I implemented as you said but it was hard to read so I divided it into three files that have a lot in common.

Reply:
Renaming hpo.py to something else seems like a good idea, at least in my opinion - it's too confusing just by looking at the name.

Reply:
I renamed it to hpo_api.py at https://github.com/openvinotoolkit/training_extensions/pull/2912/commits/57b7affda359b94f3eb8b9ad7cb74bca0671b195